mono/sgen/sgen-gc.h

```diff
@@ -909,7 +909,39 @@ struct _LOSObject {
 	GCObject data [MONO_ZERO_LEN_ARRAY];
 };
 
-extern LOSObject *sgen_los_object_list;
+#define SGEN_LOS_OBJECT_LIST_BUCKETS 31
+extern LOSObject *sgen_los_object_lists [];
+
+static inline guint
```

Comment:
uint64 is a better hash type than uint32, because, in the event of hash % bucket_count colliding, if you store the full hash (not-modded) in the table also, you can compare that for equality before doing a full compare.

Reply:
This hash function is just used to pick a bucket, no need to compare against this hash. It is also using the hash function we use elsewhere, sgen_aligned_addr_hash, but could of course use a uin64 or even do the hash % bucket_count as part of this function.